algorithms.py

```diff
@@ -851,4 +1003,4 @@ def main():
 
 
 if __name__ == "__main__":
-    main()
+    ts = main()
```

Comment:
So yes, it looks like the changes here have broken the standard coalescent code path. We need to make a separate control flow for the WF code (probably adding a command line argument --model=dtwf is the simplest). But we definitely need to be sure that the standard coalescent code path works before merging; as I say, other people are developing algorithms against the code.